### ISSUE TITLE
Let plugins (/the user) add or remove panes (so there can not just be 2 but 1..N)

### ISSUE THREAD

**Issue by mherrmann** [fman-users/fman#57]

**Comment by kek91** [comment]
@mherrmann, I just noticed this is a duplicate of #34 so I'll close this one. Let me know if I misunderstood and they have different purposes :)

**Comment by mherrmann** [comment]
Thanks @kek91 but they actually do have different purposes. This issue is meant to also allow (say) three panes _next to_ each other: left | middle | right. I'll reopen :)

**Comment by kek91** [comment]
Oh, sorry about that! I misunderstood and thought tabs/panes were the same. Thanks for the clarification :)

**Comment by mateyd** [comment]
I like the layout of http://ranger.github.io

![screenshot0](https://user-images.githubusercontent.com/37427657/37500351-552feaf0-28c9-11e8-94a8-9c3e1c5684bb.png)

Dunno if you already knew it.

**Comment by mherrmann** [comment]
@mateyd thank you pointing it out :-) Before starting fman, I compiled a list of the ~100 file managers that exist and ranger was one of them. Thanks!

**Comment by danmou** [comment]
At the moment, the lack of a single-pane mode is my main annoyance with fman (which means fman has come a long way!). I know a lot of people really like dual pane file managers, and I see the appeal for certain specific types of tasks, but 95% of the time I'm just working in the left pane and it's annoying to waste screen area on an unused pane (this is especially annoying because I'm using a tiling window manager). Moreover, I can't imagine a scenario where the benefit of more than two panes would outweigh the drawbacks, such as requiring much more complicated keybindings and API (maybe someone else has a compelling use case, but I can't think of any). Besides, you can always open up another instance of fman if you need to see more than two folders at once.

So I would suggest dropping the 3..N part of this feature and just allowing single or dual pane mode, with the possibility of switching between the two modes with a keybinding. If fman saved the position of the pane divider between sessions, then a workaround to get a single-pane mode would be to move the divider all the way to the right and leaving it there, but at the moment the position is reset when you start fman, so this doesn't work.

Edit: Just remembered there's already an issue for single pane mode (#156), but I still think it's worth reconsidering whether the additional complications of more panes is really worth it.

**Comment by mherrmann** [comment]
> At the moment, the lack of a single-pane mode is my main annoyance with fman (which means fman has come a long way!)

That's actually quite nice to hear! Glad we're on the right tracks :-)

Your point (whether more panes are really worth it) is actually quite a good one @Danmou. Maybe the people who voted for this issue actually were only interested in the case of one pane. I understand @mateyd likes the three column layout of ranger, but it includes a "Preview" pane which I'm very skeptical of as explained in eg. #385.

I'll continue with a follow-up question in #156!

**Comment by SteveBenner** [comment]
> So I would suggest dropping the 3..N part of this feature and just allowing single or dual pane mode, with the possibility of switching between the two modes with a keybinding...

I just discovered `fman` and immediately am hooked. One of the first things I thought was: this app **absolutely** needs expanded tiling options... 4 would be cool, but 1..N is where it's at, as suggested.

I do so much in terminal that multi-paned is a way of life for me now, haha!

Maybe a related feature/issue is the ability to have multiple windows open, as opposed to multiple instances of `fman`. What are your thoughts on that?

**Comment by mherrmann** [comment]
Hi Steve,

I can see it's sowewhat nicer to have multiple windows. But I feel that the benefits over multiple instances (which are already supported) are rather slim. Regarding 1..N, I also feel that's the appropriate generality.
